modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1650,6 +1650,16 @@ void GDScriptAnalyzer::resolve_function_signature(GDScriptParser::FunctionNode *
 			return_type.type_source = GDScriptParser::DataType::INFERRED;
 			return_type.kind = GDScriptParser::DataType::VARIANT;
 			p_function->set_datatype(return_type);
+#ifdef DEBUG_ENABLED
+			if (GLOBAL_GET("debug/gdscript/warnings/enforce_static_typing")) {
+				if (p_is_lambda) {
+					enforce_static_typing((GDScriptParser::Node *)p_function, "Anonymous Lambda");
+				}
+				else {
+					enforce_static_typing((GDScriptParser::Node *)p_function, p_function->identifier->name);
```

Comment:
I believe the cast is not needed since `FunctionNode` extends `Node`:

https://github.com/godotengine/godot/blob/0a7f75ec7b465604b6496c8f5f1d638aed250d6d/modules/gdscript/gdscript_parser.h#L828

Note my other comment that `enforce_static_typing()` is not needed.